### PR TITLE
2.9 Security

### LINKED ISSUE #217: 2.9 - Process Security

### PR DESCRIPTION
**Pull request by tobloef**

### CODE REVIEW

File: report/29_Process-Security.md [hunk 1]
```diff
@@ -1 +1,45 @@
...
+
+## Vulnerability Testing
+
+### SonarCube security reports
+As part of our CI pipeline, SonarCube will analyze pushed code for potential security problems, such as hard-coded credentials, out of data dependencies and insecure handling of user data. The analysis results are by no means exhaustive but serve as the first line of defense and runs almost for free, both in term of financial cost and time cost.
+
+### Penetration Testing
+Through a combination of security testing done by an internal team and an external team, the following have been discovered about the system:
+
+* The simulator API is hardcoded into the codebase. IT should instead be securely stored in an API key store or in the environment variables, even if it is currently required to be a specific known value based on the spec.
```
Comment (Duckapple) [inline report/29_Process-Security.md]:
```suggestion
* The simulator API is hardcoded into the codebase. It should instead be securely stored in an API key store or in the environment variables, even if it is currently required to be a specific known value based on the spec.
```

File: report/29_Process-Security.md [hunk 2]
```diff
@@ -1 +1,45 @@
...
+* No vulnerabilities was found on the website, likely due to how much the chosen web framework ASP.NET handles out of the box.
+  * No SQL injection, due to the use of the Entity Framework ORM.
+  * No Cross-site Request Forgery, thanks to ASP.NET’s automatic form security.
+  * No cookie stealing thanks to how ASP.NET Identity.
```
Comment (Duckapple) [inline report/29_Process-Security.md]:
```suggestion
  * No cookie stealing thanks to how ASP.NET Identity works.
```

File: report/29_Process-Security.md [hunk 3]
```diff
@@ -1 +1,45 @@
...
+  * No local data (such as cookies) in Iframes and no access so form pages such as login through Iframes. This prevents clickjacking.
```
Comment (Duckapple) [inline report/29_Process-Security.md]: "Iframes and no access so form pages such as"
Please rephrase?

**Comment by tobloef** [comment]
Closes #217

**Comment by Duckapple** [comment]
Closes #218

**Comment by jlndk** [comment]
Close #217

File: report/29_Process-Security.md [hunk 4]
```diff
@@ -1 +1,45 @@
-Brief results of the security assessment.
+# Security
```
Comment (Jakobis) [inline report/29_Process-Security.md]: /# is for the title of the document, ## is for headings of the 3 big parts of the report. ### should be the biggest heading you use

File: report/29_Process-Security.md [hunk 5]
```diff
@@ -1 +1,45 @@
...
+## Risk Assessment
+
+![Risk assessment matrix](./images/risk_matrix.png)
+
+* Leaking user data
+  * **Description**: Most user data is public, but a few critical things such as emails and passwords are not. Leaking passwords are unlikely, but still technically possible for example due to logging misconfiguration, as they are stored only as hashes. The database is the most likely attack vector for accessing sensitive data.
```
Comment (Jakobis) [inline report/29_Process-Security.md]: "Leaking passwords are unlikely, but still technically possible for example due to logging misconfiguration, as they are stored only as hashes."

Denne sætning er meget svær at læse. Burde det ikke være "Leaking passwords are unlikely as they are stored only as hashes, but it is still technically possible for example due to logging misconfiguration.

**Review by Hjaltesorgenfrei — APPROVED** [review]

**Review by Duckapple — APPROVED** [review]

**Pull request merged**